hack/overlays/webhook-configuration.yaml

```diff
@@ -21,6 +21,7 @@ metadata:
   #@overlay/match missing_ok=True
   annotations:
     cert-manager.io/inject-ca-from: cartographer-system/cartographer-webhook
+  name: cartographer-validating-webhook-configuration
```

Comment:
This will do for now, I guess, but I'm sure there's a better way. A kubebuilder/controller-runtime marker perhaps?

Reply:
didn't find a marker / way of getting around the name (it's [hardcoded](https://github.com/kubernetes-sigs/controller-tools/blob/ebdf5177ad92f89da82686f4100c06d428ab3ce5/pkg/webhook/parser.go#L384-L405)) and, _yes_, using what kubebuilder gives us (kustomize-based config gen) there is a pattern for dealing with it (e.g., see https://github.com/vmware-tanzu/cartographer-conventions/blob/35a8c0b98b2933fcfc0aeb1e97343fd78b8a9cfe/config/default/kustomization.yaml#L4-L9)

Reply:
how kubebuilder projects handle this problem by default:

https://github.com/kubernetes-sigs/kubebuilder/blob/9711dc84d0215d2ef06b8cac5046e84ae810eba2/docs/book/src/cronjob-tutorial/testdata/project/config/default/kustomization.yaml#L4-L9